### PR TITLE
Added unit tests for readMany method in AppUserCrudService.java

### LINKED ISSUE #109: Add unit tests for `readMany` method in `src/main/java/app/fitbuddy/jpa/service/crud/AppUserCrudService.java`
https://github.com/mepox/fitbuddy/blob/c3551654303f4d155a4c007f87178ece3b396af2/src/main/java/app/fitbuddy/jpa/service/crud/AppUserCrudService.java#L69-L76

Add the tests to the end of `src/test/java/app/fitbuddy/jpa/service/crud/AppUserCrudServiceTest.java`, in a new `@Nested` class, same as the other tests.

### PR DESCRIPTION
**Pull request by SwethaTamatam**
Added unit tests for readMany method in AppUserCrudService.java
Close #109

### CODE REVIEW

File: src/test/java/app/fitbuddy/jpa/service/crud/AppUserCrudServiceTest.java [hunk 1]
```diff
@@ -141,4 +144,33 @@ void readByName_whenAppUserFound_shouldReturnAppUserDto() {
 			assertEquals(appUserDtoMock, actualAppUserDto);
 		}
 	}
+	@Nested
+	class ReadMany {
+		@Test
+		void readMany_whenNoExercisesFound_shouldReturnEmptyList() {
```
Comment (mepox) [inline src/test/java/app/fitbuddy/jpa/service/crud/AppUserCrudServiceTest.java:150]: I think you meant `...whenNoUsersFound...` in the method name?

**Comment by SwethaTamatam** [comment]
Yes @mepox .Sorry, I will update and commit again.

File: src/test/java/app/fitbuddy/jpa/service/crud/AppUserCrudServiceTest.java [hunk 2]
```diff
@@ -141,4 +144,33 @@ void readByName_whenAppUserFound_shouldReturnAppUserDto() {
...
+
+		@Test
+		void readMany_whenUsersFound_shouldReturnListOfAppUserDtos() {
+			List<AppUser> appUsersMock = new ArrayList<>();
+			AppUser appUserMock = AppUserTestHelper.getMockAppUser(1, "name", "password");
+			appUsersMock.add(appUserMock);
+
+			List<AppUserDto> appUserDtosMock = new ArrayList<>();
+			AppUserDto appUserDtoMock = new AppUserDto(1, "name", "password", "roleName");
+			appUserDtosMock.add(appUserDtoMock);
```
Comment (mepox) [inline src/test/java/app/fitbuddy/jpa/service/crud/AppUserCrudServiceTest.java:166]: Of course this code works, but we could really use the `List.of()` Java feature to shorten this part. Additionally, `List.of()` returns an immutable List. As we won't need to modify the List, this act as a "safe guard" so we cannot modify the List by mistake. 👍 

```
AppUser appUserMock = AppUserTestHelper.getMockAppUser(1, "name", "password");
List<AppUser> appUsersMock = List.of(appUserMock);

AppUserDto appUserDtoMock = new AppUserDto(1, "name", "password", "roleName");
List<AppUserDto> appUserDtosMock = List.of(appUserDtoMock);
```

Then we can easily remove the unnecessary import too: `import java.util.ArrayList;`

**Review by mepox — CHANGES_REQUESTED** [review]

File: src/test/java/app/fitbuddy/jpa/service/crud/AppUserCrudServiceTest.java (hunk 2 above)
Comment (SwethaTamatam) [reply src/test/java/app/fitbuddy/jpa/service/crud/AppUserCrudServiceTest.java:166]: Sure ,will change that. Thanks for the information.

**Review by mepox — APPROVED** [review]

**Comment by mepox** [comment]
Hi @SwethaTamatam ! Thanks for your contribution and for applying the changes. The PR looks good now and it will be merged soon 🎉

**Comment by SwethaTamatam** [comment]
Thank you

**Pull request merged**